src/pages/users/[username]/edit.tsx

```diff
@@ -46,7 +46,10 @@ export const getServerSideProps = withMultipleServerSideProps(
     }
 
     if (isPost(req)) {
+      const groupsChecked = groups.filter((group) => formData[group.name] === "yes")
       const userDetails: Partial<User> = formData
+      userDetails.groups = groupsChecked
```

Comment:
(minor) you can move the groupsChecked initialisation and the initialisation of the groups from userDetails after the getUserById check? If we fail to get the user here, then the data we have slightly wasted computation to do the filtering and initialisation.

Reply:
Review comment 3